src/test/java/org/killbill/billing/plugin/util/http/TestHttpClient.java

```diff
@@ -73,6 +76,22 @@ public void handle(final HttpExchange exchange) throws IOException {
                 exchange.sendResponseHeaders(401, 0);
             }
         });
+        server.createContext("/redirect/source", exchange -> {
+            final byte[] body = "302 - will redirect".getBytes(StandardCharsets.UTF_8);
+            exchange.sendResponseHeaders(302, body.length);
+            // exchange.getResponseHeaders().add("Location", "http://" + exchange.getLocalAddress().getHostName() + ":" + exchange.getLocalAddress().getPort() + "/redirect/target");
```

Comment:
This didn't work either? Looking at the source code, it wants to create a `URI` from the String: https://github.com/openjdk/jdk/blob/a6b72f56f56b4f33ac163e90b115d79b2b844999/src/java.net.http/share/classes/jdk/internal/net/http/RedirectFilter.java#L135